### PR TITLE
Replace the 'order' for loop and conditional with a list comprehension

### PR DESCRIPTION
**Pull request by mjbear**
Make the 'order' list code more concise and readable by replacing the 'order' for loop and conditional with a list comprehension.

### CODE REVIEW

File: fnsort.py [hunk 1]
```diff
@@ -57,10 +57,7 @@ def sort_footnotes(text):
 
     # Determine the order of the footnote-links in the text. If a link is used
     # more than once, its order is its first position.
-    order = []
-    for i in links:
-        if order.count(i[1]) == 0:
-            order.append(i[1])
+    order = [i[1] for i in links]
```
Comment (hellt) [inline fnsort.py:60]: should this account for not adding duplicates? currently it doesn't in this comprehension

Comment (mjbear) [reply fnsort.py]: Agreed. My bad and glad for the review.
I will be adding an example markdown file with duplicate tag(s).
(I should prioritize the unit testing :grin: - especially given a situation like this that it should have caught.)

I made an update and my testing says the modified comprehension should be good now.

Thank you!

**Comment by hellt** [comment]
lets add a test for this one in this PR to make sure it works

**Comment by mjbear** [comment]
> lets add a test for this one in this PR to make sure it works

Understood.

What I'll do is get some unit testing plumbed up (based off the `main` branch). From there we can review the testing PR and (if all goes well) merge the unit testing PR into your main branch. At that point I'll rebase the feature branch for this PR off of the updated position of the main branch to do sync up and perform testing.

**Comment by mjbear** [comment]
> > lets add a test for this one in this PR to make sure it works
> 
> Understood.
> 
> What I'll do is get some unit testing plumbed up (based off the `main` branch). From there we can review the testing PR and (if all goes well) merge the unit testing PR into your main branch. At that point I'll rebase the feature branch for this PR off of the updated position of the main branch to do sync up and perform testing.

Submitted PR #5 that adds unit testing

**Comment by mjbear** [comment]
@hellt 
I rebased this branch against `main` so it is current.
The Python unit tests succeed! :grinning:

Thank you for giving this PR a look. :wink:

**Comment by hellt** [comment]
Thanks!

**Pull request merged**